configurator/src/main.rs

```diff
@@ -350,9 +408,89 @@ fn main() -> Result<(), anyhow::Error> {
                         masked: false,
                     },
                 },
+                rpc: match config.rpc.enabled {
+                    true => Some(RpcProperties {
+                        quick_connect_url: Property::String {
+                            value: format!(
+                                "clightning-rpc://{}:{}@{}:{}",
+                                config.rpc.user, config.rpc.password, peer_tor_address, 8080
+                            ),
+                            description: Some(
+                                "A convenient way to connect a wallet to a remote node".to_owned(),
+                            ),
+                            copyable: true,
+                            qr: true,
+                            masked: true,
+                        },
+                        rpc_username: Property::String {
+                            value: format!("{}", config.rpc.user,),
+                            description: Some("Username for RPC connections".to_owned()),
+                            copyable: true,
+                            qr: false,
+                            masked: true,
+                        },
+                        rpc_password: Property::String {
+                            value: format!("{}", config.rpc.password,),
+                            description: Some("Password for RPC connections".to_owned()),
+                            copyable: true,
+                            qr: false,
+                            masked: true,
+                        },
+                    }),
+                    false => None,
+                },
+                rest: match config.advanced.plugins.rest {
+                    true => Some(RestProperties {
+                        rest_port: Property::String {
+                            value: format!(
+                                "{}",
+                                3001,
+                            ),
+                            description: Some(
+                                "The port your c-lightning-REST API is listening on"
+                                    .to_owned(),
+                            ),
+                            copyable: true,
+                            qr: false,
+                            masked: false,
+                        },
```

Comment:
you may wish to create utility functions to compress the amount of boilerplate you have to do. not sure if it will help or not.